index.ts

```diff
@@ -468,5 +500,82 @@ export class HttpClient {
         retryNumber = Math.min(ExponentialBackoffCeiling, retryNumber);
         const ms: number = ExponentialBackoffTimeSlice*Math.pow(2, retryNumber);
         return new Promise(resolve => setTimeout(()=>resolve(), ms));
-    } 
+    }
+
+    private static dateTimeDeserializer(key: any, value: any): any {
+        if (typeof value === 'string'){
+            let a = new Date(value);
+            if (!isNaN(a.valueOf())) {
+                return a;
+            }
+        }
+
+        return value;
+    }
+
+    private async _processResponse<T>(res: ifm.IHttpClientResponse, options: ifm.IRequestOptions): Promise<ITypedResponse<T>> {
+        return new Promise<ITypedResponse<T>>(async (resolve, reject) => {
+            const statusCode: number = res.message.statusCode;
+
+            const response: ITypedResponse<T> = {
+                statusCode: statusCode,
+                result: null,
+                headers: {}
+            };
+
+            // not found leads to null obj returned
+            if (statusCode == HttpCodes.NotFound) {
+                resolve(response);
+            }
+
+            let obj: any;
+            let contents: string;
+
+            // get the result from the body
+            try {
+                contents = await res.readBody();
+                if (contents && contents.length > 0) {
+                    if (options && options.deserializeDates) {
+                        obj = JSON.parse(contents, HttpClient.dateTimeDeserializer);
```

Comment:
> dateTimeDeserializer

Feels gross since it's not based on the metadata of the type. I guess that's the way it has to be in js. And opt-in so whatevs :)

For my own benefit, curious about the specific use case?

Reply:
yep.  this carried over from the typed rest client